src/main.rs

```diff
@@ -1,21 +1,99 @@
-use actix_web::{get, App, HttpResponse, HttpServer, Responder};
+use actix_web::{
+    error, get,
+    http::{header::ContentType, StatusCode},
+    web, App, HttpResponse, HttpServer, Responder, Result,
+};
+use derive_more::{Display, Error};
 use lazy_static::lazy_static;
 use rand::seq::SliceRandom;
 use rand::thread_rng;
+use serde::Serialize;
 use std::collections::HashMap;
 
-const NUMBER_OF_QUESTIONS: usize = 5;
+#[derive(Debug, Display, Error)]
+enum AcerError {
+    #[display(fmt = "internal error")]
+    InternalError,
+    #[display(fmt = "bad request")]
+    BadClientData,
+    #[display(fmt = "timeout")]
+    Timeout,
+}
+
+impl error::ResponseError for AcerError {
+    fn error_response(&self) -> HttpResponse {
+        HttpResponse::build(self.status_code())
+            .insert_header(ContentType::html())
+            .body(self.to_string())
+    }
+
+    fn status_code(&self) -> StatusCode {
+        match *self {
+            AcerError::InternalError => StatusCode::INTERNAL_SERVER_ERROR,
+            AcerError::BadClientData => StatusCode::BAD_REQUEST,
+            AcerError::Timeout => StatusCode::GATEWAY_TIMEOUT,
+        }
+    }
+}
+
+#[derive(Serialize, Clone)]
+struct QuestionAnswer {
+    question: String,
+    answer: String,
+}
 
 lazy_static! {
-    static ref QUESTION_KV_DATABASE: HashMap<u32, (&'static str, &'static str)> = {
+    static ref QUESTION_KV_DATABASE: HashMap<u32, QuestionAnswer> = {
         let mut m = HashMap::new();
-        m.insert(0, ("What's my name?", "Acer\n"));
-        m.insert(1, ("What's Luiz0s name?", "Luiz\n"));
-        m.insert(2, ("What's Jorge's name?", "GOD\n"));
-        m.insert(3, ("Are you sentient?", "On Ma I toN\n"));
-        m.insert(4, ("Where do you live?", "Stalker!\n"));
-        m.insert(5, ("What's the purpose of life?", "FU\n"));
-        m.insert(6, ("How to make a good question?", "That's not the way!\n"));
+        m.insert(
```

Comment:
Maybe use HashMap::from() instead of inserting.

Reply:
great ty !!!